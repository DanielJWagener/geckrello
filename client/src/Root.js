// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
