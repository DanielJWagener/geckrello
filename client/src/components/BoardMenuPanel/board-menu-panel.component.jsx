import React from "react";

import "./board-menu-panel.styles.scss";

const BoardMenuPanel = ({ visible, togglePanel, children, heading }) => (
  <div
    className={`boardmenu__panel boardmenu__panel--${
      visible ? "visible" : "hidden"
    }`}
  >
    <h2 className="boardmenu__heading">{heading}</h2>
    <div
      className="boardmenu__panel--close"
      onClick={() => togglePanel("none")}
    >
      &times;
    </div>
    {children}
  </div>
);

export default BoardMenuPanel;
