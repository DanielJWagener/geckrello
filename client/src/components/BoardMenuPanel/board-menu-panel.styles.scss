@import "../../sass/abstracts/_variables.scss";

.boardmenu__panel {
  position: absolute;
  top: 0;
  width: $sidebar-width;

  height: calc(var(--vh, 1vh) * 100 - 9rem);
  overflow-y: auto;
  background-color: white;
  border-left: 2px solid $component-background;
  border-radius: 8px 0 0 8px;
  padding: 1.5rem;
  transition: all 0.3s;

  @media only screen and (max-width: 400px) {
    width: 100vw;
  }

  &--hidden {
    right: calc(-1 * #{$sidebar-width});
  }

  &--visible {
    right: 0;
  }

  &--close {
    position: absolute;
    top: 3px;
    right: 1rem;
    cursor: pointer;
    font-size: 3rem;

    &:hover {
      color: red;
    }
  }
}

.boardmenu__heading {
  font-size: 2rem;
  font-weight: 700;
  text-align: center;
  margin-bottom: 1.5rem;
}
