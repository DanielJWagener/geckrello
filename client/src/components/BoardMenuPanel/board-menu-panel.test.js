import { shallow } from "enzyme";
import React from "react";
import BoardMenuPanel from "./board-menu-panel.component";
import Root from "../../Root";

it("renders BoardMenuPanel component", () => {
  expect(
    shallow(
      <Root>
        <BoardMenuPanel />
      </Root>
    )
  ).toMatchSnapshot();
});
