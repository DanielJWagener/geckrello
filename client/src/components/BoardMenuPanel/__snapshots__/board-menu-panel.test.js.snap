// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardMenuPanel component 1`] = `ShallowWrapper {}`;
